apps/openglcompute/jni/oglc_run.cpp

```diff
@@ -0,0 +1,256 @@
+#include <android/log.h>
+#include <jni.h>
+#include <iostream>
+#include <iomanip>
+#include "avg_filter_uint32t.h"
+#include "avg_filter_uint32t_arm.h"
+#include "avg_filter_float.h"
+#include "avg_filter_float_arm.h"
+#include <sstream>
+
+#define  LOGI(...)  __android_log_print(ANDROID_LOG_INFO, "oglc_run", __VA_ARGS__)
+#define  LOGE(...)  __android_log_print(ANDROID_LOG_ERROR, "oglc_run", __VA_ARGS__)
+
+extern "C" int halide_copy_to_host(void *, buffer_t *);
+extern "C" int halide_device_sync(void *, buffer_t *);
+
+typedef int (filter_t) (buffer_t *, buffer_t *);
```

Comment:
There's a weird mix here of function pointers types vs. function types below... I think it all works out to the same thing (a function type is implicitly a function pointer in C...?) but it's a bit confusing. I usually typedef a function pointer type, and then just use that as a pointer type (no extra indirection) everywhere.